### ISSUE TITLE
[DICT] Add Python support for pop

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#1886]
### Describe the feature
Syntactic and semantic support should be added for `pop`. This should be tested by implementing Python printing

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
def foo():
    a = {1:1.0, 2:2.0}
    b = a.pop(1)
    c = a.pop(3, 4.0)
    return a, b, c
```

**Comment by learning-dev** [comment]
Hi @EmilyBourne, can I please take this up? Also, any reference that I can look up and fix it? I am new to OSS. Thanks.

**Comment by EmilyBourne** [comment]
> Hi @EmilyBourne, can I please take this up? Also, any reference that I can look up and fix it? I am new to OSS. Thanks.

Hi @learning-dev, thanks for your interest in Pyccel. This issue has actually already been fixed (see #1891). We just have a long merge queue at the moment as the person with merge rights is not available.

There are several similar issues though if you are interested? For example issue #1885 should be very similar. There is a longer description of how to tackle this kind of issue here: https://github.com/pyccel/pyccel/issues/1887#issuecomment-2126417978

**Comment by learning-dev** [comment]
Thank you @EmilyBourne .
